File: src/main/java/sensormodels/activity/ActivitySensorBuilder.java
package sensormodels.activity;

public interface ActivitySensorBuilder {

  ActivitySensorBuilder setSensorName(String sensorName);

  ActivitySensorBuilder setTimeStamp(String timeStamp);

  ActivitySensorBuilder setTimestamp(String timeStamp);

  ActivitySensorBuilder setSensorData(ActivitySensorData.SensorData sensorData);

  ActivitySensorBuilder setStepCounts(Integer stepCounts);

  ActivitySensorBuilder setStepDelta(Integer stepDelta);

  ActivitySensorData build();
}

File: src/main/java/sensormodels/activity/ActivitySensorBuilder.kt
package sensormodels.activity

interface ActivitySensorBuilder {
    fun setSensorName(sensorName: String?): ActivitySensorBuilder
    fun setTimeStamp(timeStamp: String?): ActivitySensorBuilder
    fun setTimestamp(timeStamp: String?): ActivitySensorBuilder
    fun setSensorData(sensorData: ActivitySensorData.SensorData?): ActivitySensorBuilder
    fun setStepCounts(stepCounts: Int?): ActivitySensorBuilder
    fun setStepDelta(stepDelta: Int?): ActivitySensorBuilder
    fun build(): ActivitySensorData
}